packages/generator/src/generator.spec.ts

```diff
@@ -226,6 +234,109 @@ describe('generator', () => {
     });
   });
 
+  describe('optionsPerService', () => {
+    beforeEach(async () => {
+      mock({
+        common: {},
+          // '/prettier/config': JSON.stringify({ printWidth: 66 }),
+          [pathTestResources]: mock.load(pathTestResources),
+          [pathToGeneratorCommon]: mock.load(pathToGeneratorCommon),
+          [pathRootNodeModules]: mock.load(pathRootNodeModules),
+          existingConfig:
+          '{ "API_TEST_SRV": {"directoryName": "test-service" } }',
+          anotherConfig:
+          '{ "inputDir/spec2.json": {"directoryName": "customName" } }'
+        });
+    });
+
+    afterEach(() => {
+      mock.restore();
+    });
+
+    it('writes options per service', async () => {
+    const options = createOptions({
+      inputDir: pathTestService,
+      outputDir: 'out',
+      optionsPerService: 'test-service-options.json',
+      skipValidation: true,
+      overwrite: true
+    });
+      await generate(options);
+
+      const actual = readFile('test-service-options.json', 'utf8');
+      await expect(actual).resolves.toMatch(
+        JSON.stringify(
+          {
+            'API_TEST_SRV': {
+              directoryName: 'test-service',
+              servicePath: '/sap/opu/odata/sap/API_TEST_SRV',
+              npmPackageName: 'test-service'
+            }
+          },
+          null,
+          2
+        )
+      );
+    });
+
+    xit('overwrites writes options per service', async () => {
```

Comment:
[q] these are the not working thing right which work in OpenApi - add a TODO and mention the lines in the ticket - TDD :-)